### PR TITLE
Adding templates to Issue section

### PR DESCRIPTION
**Pull request by kannankvs**
##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        Ansible 2.12
+    validations:
+      required: true
+  - type: input
+    id: Productv 
+    attributes:
+      label: Product Version
+      description: Enter the Product version. 
+      placeholder: |
+        eSONiC product verion
```
Comment (kerry-meyer) [inline .github/ISSUE_TEMPLATE/bug_report.yml:53]: Typo: "verion" => "version"

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      description: Enter the Ansible Collection version.
+      placeholder: |
+        Ansible 2.12
+    validations:
+      required: true
+  - type: input
+    id: Productv 
+    attributes:
+      label: Product Version
+      description: Enter the Product version.
```
Comment (kerry-meyer) [inline .github/ISSUE_TEMPLATE/bug_report.yml:51]: It would be useful to specify the command that provides the version information in the description.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
- I have provided inline comments for the bug report templates. (The other two templates look fine to me.) Most of the comments posted for the bug report ".md" template also apply to the bug report ".yml" template. Please check the corresponding sections in the .yml file even if I did not explicitly post a second copy of the comment in the matching section of that file.

- Also, just a question: Why are the "bug report" templates provided in both ".yml" and ".md" format? If there is a reason for doing this, it's fine, but the names need to be different (as suggested in the inline comments in the bug_report.yml file).

- In order to make the templates usable when creating issues, there needs to also be at least a minimal (one line) "config.yml" file in the ISSUE_TEMPLATE directory. Please add one for this PR.

    REF: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
+---
+name: 🐛 Bug report
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml:2]: @kerry-meyer  As suggested, bug_report.md file has been removed and bug_report.yml has been updated as per the suggestions mentioned for both .md file and .yml file. And a separate PR will be raised for "config.yml" file.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+<!--- HINT: enterprise sonic,SFSS, SFS, etc. -->
+
+
+##### Ansible Collection Version
+<!--- Paste verbatim command output below. -->
+```paste below
+
+```
+
+##### Product Version
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+<!--- Enter the Product version. -->
+<!--- HINT: eSONiC product verion -->
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+##### Configuration
+<!--- Paste verbatim output from ansible-config dump --only-changed -->
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+name: 🐛 Bug report
+about: Create an issue to help us improve.
+---
+
+##### Bug Description
+<!--- Write a brief summary of the issue, including rationale and design decisions. -->
+
+
+##### Product Name
+<!--- Write the short name of the Product. -->
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+
+##### Component Name
+<!--- Write the short name of the module, plugin, task, or feature. -->
+
+<!--- HINT: enterprise sonic,SFSS, SFS, etc. -->
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+
+##### Ansible Collection Version
+<!--- Paste verbatim command output below. -->
+```paste below
+
+```
+
+##### Product Version
+<!--- Enter the Product version. -->
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Addressed in .yml file as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+      placeholder: sonic_bgp, sonic_ntp, sonic_interfaces, etc.
+    validations:
+      required: true
+  - type: input
+    id: ansible
+    attributes:
+      label: Ansible Collection Version
+      description: Enter the Ansible Collection version.
+      placeholder: |
+        Ansible 2.12
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Addressed as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+      label: Component or Module Name
+      description: Write the short name of the module, plugin, task or feature below, use your best guess if unsure
+      placeholder: sonic_bgp, sonic_ntp, sonic_interfaces, etc.
+    validations:
+      required: true
+  - type: input
+    id: ansible
+    attributes:
+      label: Ansible Collection Version
+      description: Enter the Ansible Collection version.
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Addressed as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+      description: Enter the Product version. 
+      placeholder: |
+        eSONiC product verion
+    validations:
+      required: true
+  - type: textarea
+    id: device
+    attributes:
+      label: Configuration
+      description: Paste verbatim output from ansible-config dump --only-changed
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Addressed as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Addressed.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Addressed as suggested.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml:2]: Is there a reason not to at least provide a stub config.yml file with this PR? (Without it, the template isn't usable.)

[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml:2]: Resolved.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Resolved.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Resolved.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Resolved.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Resolved.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Most of the issues are resolved now.

I have posted a comment requesting a minor, one-line correction.

I have also posted a question regarding the "config.yml" file.

**Comment by kerry-meyer** [comment]
To facilitate tracking of issues, I have marked as "Resolved" the issue where Kannan proposed, as a side-note, deferring the addition of a "config.yml" file for this PR. I marked that issue "Resolved" because it pertained primarily to a resolved issue other than the "config.yml" file.

To provide a visible instance of my pending comment on the "config.yml" file, I am re-posting it here. (See the bottom of the quoted section below.)

Quoting:

```
@[kannankvs](https://github.com/kannankvs) kannankvs [Nov 7, 2022](https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/123#discussion_r1016166248)

@kerry-meyer As suggested, bug_report.md file has been removed and bug_report.yml has been updated as per the suggestions mentioned for both .md file and .yml file. And a separate PR will be raised for "config.yml" file.
Collaborator
@[kerry-meyer](https://github.com/kerry-meyer) kerry-meyer [Nov 9, 2022](https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/123#discussion_r1017570400)

Is there a reason not to at least provide a stub config.yml file with this PR? (Without it, the template isn't usable.)
```

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 12 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Thanks for doing the push yesterday to resolve this issue.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+      description: Provide the first line of "show version" output. 
+      placeholder: |
+        SONiC Software Version
+    validations:
+      required: true
+  - type: textarea
+    id: device
+    attributes:
+      label: Configuration
+      description: Paste verbatim output from ansible-config dump "--only-changed"
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Fixed with the most recent commit to this PR branch.

Resolved.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 14]
```diff
@@ -0,0 +1,2 @@
+blank_issues_enabled: false
+
```
Comment (kerry-meyer) [inline .github/ISSUE_TEMPLATE/config.yml:2]: Thanks for adding this stub.

There is a blank line at the end of the file that is causing a sanity error. Please delete the blank line to eliminate the sanity error. That will clear the way for approving and merging this PR.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thanks for making the changes to resolve the remaining issues.

There is a blank line at the end of the ".github/ISSUE_TEMPLATE/config.yml file that is causing an Ansible sanity failure. Please delete this line. That should clear the way for approval and merging of this PR.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 14 above)
Comment (kannankvs) [reply .github/ISSUE_TEMPLATE/config.yml]: @kerry-meyer Have fixed the issue on sanity error. And the checks have passed

Comment (kerry-meyer) [reply .github/ISSUE_TEMPLATE/config.yml]: Thanks for this change and all of your work on the templates.

I will mark this issue "Resolved" and will approve and merge the content for this PR.

**Review by kerry-meyer — APPROVED** [review]
Thank you for providing these templates and working with us to incorporate suggested modifications.

Approved.

**Pull request merged**